error: Could not compile `ethcore-network-devp2p` when trying to install with Rust Nightly. error: multiple packages with binaries found: polkadot, subkey when running install command without appending the binaries

When I followed the [To play](https://github.com/paritytech/polkadot#to-play) instructions to install Polkadot PoC-1 by running `cargo install --git https://github.com/paritytech/polkadot.git --branch v0.1` it returned the following error because I didn't follow the instruction to switch to "Rust (1.26.1 stable)" as instructed, and I was actively using Rust 1.27.0 Nightly (i.e. `rustc --version` returned rustc 1.27.0-nightly (ad610bed8 2018-04-11)) instead of the Stable version:
```
    Updating git repository `https://github.com/paritytech/polkadot.git`
  Installing polkadot v0.1.0 (https://github.com/paritytech/polkadot.git?branch=v0.1#28cb780b)
   Compiling libc v0.2.36
   ...
   Compiling ethcore-network-devp2p v1.12.0 (https://github.com/paritytech/parity.git#0ecbb3ec)
error[E0283]: type annotations required: cannot resolve `std::string::String: std::convert::AsRef<_>`
  --> /Users/Ls/.cargo/git/checkouts/parity-dc9825eb65b3adf1/0ecbb3e/util/network-devp2p/src/service.rs:35:91
   |
35 |                 info!(target: "network", "Public node URL: {}", Colour::White.bold().paint(public_url.as_ref()));
   |                                                                                                       ^^^^^^

error: aborting due to previous error

For more information about this error, try `rustc --explain E0283`.
error: Could not compile `ethcore-network-devp2p`.
warning: build failed, waiting for other jobs to finish...
error: failed to compile `polkadot v0.1.0 (https://github.com/paritytech/polkadot.git?branch=v0.1#28cb780b)`, intermediate artifacts can be found at `/var/folders/8q/kll0jy354kj5gxrv9n0pwx300000gn/T/cargo-installrap3F9`
```

I installed the latest stable version of Rust Stable with `rustup install stable` (which installed rustc 1.26.2)

I proceeded to update to the latest versions of Rust Stable and Rust Nightly by running `rustup update`, which returned:
```
   stable-x86_64-apple-darwin updated - rustc 1.26.2 (594fb253c 2018-06-01)
  nightly-x86_64-apple-darwin updated - rustc 1.28.0-nightly (4a9c58c6b 2018-06-05)
```

I changed the active toolchain to the latest Rust Stable version with `rustup default stable` (or `rustup default 1.26.2`)

I checked what active toolchain I was using by running `rustup show`
```
Default host: x86_64-apple-darwin

installed toolchains
--------------------

stable-x86_64-apple-darwin
nightly-x86_64-apple-darwin
1.26.1-x86_64-apple-darwin
1.26.2-x86_64-apple-darwin (default)

active toolchain
----------------

1.26.2-x86_64-apple-darwin (default)
rustc 1.26.2 (594fb253c 2018-06-01)
```

Also, the [To play](https://github.com/paritytech/polkadot#to-play) instructions also state that "You can drop the --branch v0.1 to get the very latest version of Polkadot, but these instructions might not work in that case.". So I tried running `cargo install --git https://github.com/paritytech/polkadot.git`, but it returned the following error:
```
    Updating git repository `https://github.com/paritytech/polkadot.git`
error: multiple packages with binaries found: polkadot, subkey
```
I then resolved it by running the following instead `cargo install --git https://github.com/paritytech/polkadot.git polkadot subkey`, which installed Polkadot PoC-1 successfully, so I could then proceed to trying it out with `polkadot --chain=dev --validator --key Alice` successfully too.

Conclusions:
* Polkadot PoC-1 appears only install when latest Rust Stable (i.e. 1.26.1 or 1.26.2) is the active toolchain by first running `rustup update`, `rustup default stable`, and then installing it with `cargo install --git https://github.com/paritytech/polkadot.git polkadot subkey` (i.e. it was necessary to append `polkadot subkey` after dropping `--branch v0.1` from the command)
* Polkadot PoC-1 does not install when a Rust Nightly version is the active toolchain

My System:
* macOS 10.13.4

--- a/README.md
+++ b/README.md
@@ -5,7 +5,7 @@ Implementation of a https://polkadot.io node in Rust.
 ## To play
 
 If you'd like to play with Polkadot, you'll need to install a client like this
-one. First, get Rust (1.26.1 or 1.26.2 stable) and the support software if you don't already have it, and ensure you change the active toolchain to the stable version instead of the nightly version:
+one. First, get Rust (1.26.1 or later) and the support software if you don't already have it, and ensure you change the active toolchain to the stable version instead of the nightly version:
 
 ```
 curl https://sh.rustup.rs -sSf | sh

--- a/README.md
+++ b/README.md
@@ -5,11 +5,12 @@ Implementation of a https://polkadot.io node in Rust.
 ## To play
 
 If you'd like to play with Polkadot, you'll need to install a client like this
-one. First, get Rust (1.26.1 stable) and the support software if you don't already have it:
+one. First, get Rust (1.26.1 or 1.26.2 stable) and the support software if you don't already have it, and ensure you change the active toolchain to the stable version instead of the nightly version:
 
 ```
 curl https://sh.rustup.rs -sSf | sh
 sudo apt install make clang
+rustup default stable
 ```
 
 Then, install Polkadot PoC-1:
@@ -19,7 +20,7 @@ cargo install --git https://github.com/paritytech/polkadot.git --branch v0.1
 ```
 
 You'll now have a `polkadot` binary installed to your `PATH`. You can drop the
-`--branch v0.1` to get the very latest version of Polkadot, but these
+`--branch v0.1` or run `cargo install --git https://github.com/paritytech/polkadot.git polkadot subkey` to get the very latest version of Polkadot, but these
 instructions might not work in that case.
 
 ### Development

